### PR TITLE
Remove static methods from widgets in widget_toolkit

### PR DESCRIPTION
**Pull request by DDavidPrime**

### CODE REVIEW

File: packages/widget_toolkit/example/macos/Flutter/ephemeral/Flutter-Generated.xcconfig [hunk 1]
```diff
@@ -1,6 +1,6 @@
 // This is a generated file; do not edit or check into version control.
-FLUTTER_ROOT=/opt/homebrew/Caskroom/flutter/3.0.0/flutter
-FLUTTER_APPLICATION_PATH=/Users/stanislav.petkov/StudioProjects/widget_toolkit/packages/widget_toolkit/example
+FLUTTER_ROOT=/Users/pwndp/fvm/versions/3.22.2
+FLUTTER_APPLICATION_PATH=/Users/pwndp/repos/widget_toolkit/packages/widget_toolkit/example
```
Comment (StanevPrime) [inline packages/widget_toolkit/example/macos/Flutter/ephemeral/Flutter-Generated.xcconfig:3]: Do we really need to add user specific configurations such as `/Users/pwndp` ?

File: packages/widget_toolkit/example/linux/flutter/ephemeral/.plugin_symlinks/url_launcher_linux [hunk 2]
```diff
@@ -0,0 +1 @@
+/Users/pwndp/.pub-cache/hosted/pub.dev/url_launcher_linux-3.1.1/
```
Comment (StanevPrime) [inline packages/widget_toolkit/example/linux/flutter/ephemeral/.plugin_symlinks/url_launcher_linux:1]: Do we really need to add user specific configurations such as /Users/pwndp ?

File: packages/widget_toolkit/example/macos/Flutter/ephemeral/flutter_export_environment.sh [hunk 3]
```diff
@@ -1,7 +1,7 @@
 #!/bin/sh
 # This is a generated file; do not edit or check into version control.
-export "FLUTTER_ROOT=/opt/homebrew/Caskroom/flutter/3.0.0/flutter"
-export "FLUTTER_APPLICATION_PATH=/Users/stanislav.petkov/StudioProjects/widget_toolkit/packages/widget_toolkit/example"
+export "FLUTTER_ROOT=/Users/pwndp/fvm/versions/3.22.2"
```
Comment (StanevPrime) [inline packages/widget_toolkit/example/macos/Flutter/ephemeral/flutter_export_environment.sh:3]: Do we really need to add user specific configurations such as /Users/pwndp ?

File: packages/widget_toolkit/lib/src/lib_item_picker/views/item_picker_page_with_dependencies.dart [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  final String? title;
+  final String? saveButtonText;
+  final WidgetBuilder? footerBuilder;
+  final ItemPickerItemBuilder<T>? itemBuilder;
+  final Widget Function(Exception)? errorBuilder;
+  final Widget Function()? emptyBuilder;
+  final Widget Function(BuildContext context, int index)? separatorBuilder;
+
+  @override
+  Widget build(BuildContext context) => MultiProvider(
```
Comment (StanevPrime) [inline packages/widget_toolkit/lib/src/lib_item_picker/views/item_picker_page_with_dependencies.dart:47]: @nikolay-vasilev-prime here is one example where`isInTestMode` should be integrated to enable the golden tests capabilities

File: packages/widget_toolkit/lib/src/lib_open_url/views/open_url_widget_with_dependencies.dart [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  final String url;
+  final Widget child;
+  final VoidCallback? onTap;
+  final Function(BuildContext, Exception)? onError;
+  final String Function(BuildContext, Exception)? translateError;
+  final UriType uriType;
+  final ModalConfiguration modalConfiguration;
+
+  @override
+  Widget build(BuildContext context) => MultiProvider(
```
Comment (StanevPrime) [inline packages/widget_toolkit/lib/src/lib_open_url/views/open_url_widget_with_dependencies.dart:27]: @nikolay-vasilev-prime here is one example whereisInTestMode should be integrated to enable the golden tests capabilities

File: packages/widget_toolkit/lib/src/lib_search_picker/views/search_picker_page_with_dependencies.dart [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+  final ItemPickerItemBuilder<T>? itemBuilder;
+  final T? selectedItem;
+  final Function(T?)? onItemTap;
+  final Widget Function(Exception)? errorBuilder;
+  final Widget Function()? emptyBuilder;
+  final Widget Function(int)? separatorBuilder;
+  final bool showEmptyWidgetWhenNoResultsAreFound;
+
+  @override
+  Widget build(BuildContext context) => MultiProvider(
```
Comment (StanevPrime) [inline packages/widget_toolkit/lib/src/lib_search_picker/views/search_picker_page_with_dependencies.dart:40]: @nikolay-vasilev-prime here is one example whereisInTestMode should be integrated to enable the golden tests capabilities

**Review by StanevPrime — CHANGES_REQUESTED** [review]

**Review by StanevPrime — APPROVED** [review]

File: packages/widget_toolkit/example/macos/Flutter/ephemeral/Flutter-Generated.xcconfig (hunk 1 above)
Comment (DDavidPrime) [reply packages/widget_toolkit/example/macos/Flutter/ephemeral/Flutter-Generated.xcconfig]: Done!

File: packages/widget_toolkit/example/linux/flutter/ephemeral/.plugin_symlinks/url_launcher_linux (hunk 2 above)
Comment (DDavidPrime) [reply packages/widget_toolkit/example/linux/flutter/ephemeral/.plugin_symlinks/url_launcher_linux]: Done!

File: packages/widget_toolkit/example/macos/Flutter/ephemeral/flutter_export_environment.sh (hunk 3 above)
Comment (DDavidPrime) [reply packages/widget_toolkit/example/macos/Flutter/ephemeral/flutter_export_environment.sh]: Not sure why those files/paths were included with this PR, but I've listed them in gitignore.

**Pull request merged**